src/Model/TransactionalManagerInterface.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\Doctrine\Model;
+
+/**
+ * @author Erison Silva <erison.sdn@gmail.com>
+ */
+interface TransactionalManagerInterface
```

Comment:
it may sens to extends ManagerInterface since there is no reason to implements this one without the Transactional one.

Also, I saw the ClearableManagerInterface does not extends the ManagerInterface. I think we should add a NEXT_MAJOR to do it.

Reply:
Hey @VincentLanglet if I do this BaseManager will be forced to implements the Transaction methods, and as you said above for ODM there isn't those fields!

Reply:
No, I meant we should have

TransactionalManagerInterface extends ManagerInterface
ClearableManagerInterface extends ManagerInterface

BaseManager implements ManagerInterface, ClearableManagerInterface
BaseEntityManager extends BaseManager implements TransactionalManagerInterface

Reply:
> No, I meant we should have
> 
> TransactionalManagerInterface extends ManagerInterface ClearableManagerInterface extends ManagerInterface
> 
> BaseManager implements ManagerInterface, ClearableManagerInterface BaseEntityManager extends BaseManager implements TransactionalManagerInterface

I disagree, I would keep Transaction just with the methods that it's important for transaction, find, findBy, findAll and so on, is not required for Transaction works!

and keep this separated, we have the flexibility to Manager the transaction in one separated class in case we want!

Reply:
Indeed, there is too much in the ManagerInterface.

Best would be to have a smaller interface with just save and delete maybe. and the transactional will only extends this smaller interface.

My idea was that it make no sens to have transaction/commit method without a save/delete method.
But it's maybe to much to create another interface just for this.

We can keep it like this.

Reply:
I don't think it's too much to create interface with `save` and `delete`, even if it's only one method

Look this example: https://github.com/symfony/symfony/blob/6.2/src/Symfony/Component/Security/Http/EntryPoint/AuthenticationEntryPointInterface.php

Symfony created this `AuthenticationEntryPointInterface` with only one method, It gives me the flexibility

to implement the start entry point directly in my Authenticator, or create just a class with this entry point and uses this in 2 or more authenticators, it was my case! I have more Then one authenticator using the same `start point`!

Reply:
We should decide now IMHO.

If I'm right, the ClearableManagerInterface should extends the ManagerInterface because calling `clear` may have impact on the previously found entity (I think the manager doesn't know them anymore so you have to find them again).
So it makes sens that ClearableManagerInterface has ManagerInterface in the name.

If we decide to create a small interface with save and delete (let's call SmallInterface for now) and make TransactionalManagerInterface extends SmallInterface, it won't make sens having ManagerINterface in the name. It should be more TransactionalSmallInterface.

There is three solutions
- TranslationalManagerInterface extends nothing, lot of flexibility even if can be weird
- TransactionalManagerINterface extends ManagerINtefarce, easier for us, but less flexibility
- Creating a smaller managerInterface and extending it ; it require some reflexion and a name.

I may be lazy and will accept the solution 1...
@jordisala1991 Wdyt we should do ?

Reply:
solution 1